realm/realm-library/src/main/java/io/realm/RealmCache.java

```diff
@@ -117,6 +119,19 @@ private RealmCache(RealmConfiguration config) {
 
         RefAndCount refAndCount = cache.refAndCountMap.get(RealmCacheType.valueOf(realmClass));
 
+        if (refAndCount.globalCount == 0) {
+            SharedRealm sharedRealm = SharedRealm.getInstance(configuration);
+            if (Table.primaryKeyTableNeedsMigration(sharedRealm)) {
```

Comment:
Why add the extra method to check if the migration is needed? Couldn't you just call migrationIfNeeded - it doesn't do anything unless needed? An empty transaction wouldn't hurt?